common/src/db/query.rs

```diff
@@ -130,42 +145,210 @@ pub enum Error {
 // Internal types
 /////////////////////////////////////////////////////////////////////////
 
+#[derive(Default)]
+pub struct Columns {
+    columns: Vec<(ColumnRef, ColumnDef)>,
+}
```

Comment:
If I'm understanding correctly, you're expanding the public interface of this module to include `Columns`, so the comment above it is wrong, since "internal types" aren't public.